### PR TITLE
Fix unused Update parameter

### LINKED ISSUE #56740: `StackAllocArrayCreationExpressionSyntax.Update` does not account for the `stackAllocKeyword` parameter
[Source](https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Syntax/StackAllocArrayCreationExpressionSyntax.cs#L11-L12)

This appears to be unintended.

### PR DESCRIPTION
**Pull request by AlFasGD**
Fixes #56740

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Syntax/StackAllocArrayCreationExpressionSyntax.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Syntax
     public partial class StackAllocArrayCreationExpressionSyntax
     {
         public StackAllocArrayCreationExpressionSyntax Update(SyntaxToken stackAllocKeyword, TypeSyntax type)
-            => Update(StackAllocKeyword, type, Initializer);
+            => Update(stackAllocKeyword, type, Initializer);
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Syntax/StackAllocArrayCreationExpressionSyntax.cs:12]: If you wouldn't mind, I think we'd like to have a test case to make sure this doesn't regress in the future.

Comment (AlFasGD) [reply src/Compilers/CSharp/Portable/Syntax/StackAllocArrayCreationExpressionSyntax.cs:12]: This sounds like even more test syntax generation.

I only found this through IDE0020, which did not appear anywhere else. The existing tests for the auto-generated `Update` functions don't include the hand-written ones.

Adding a test for this specific function doesn't prevent any future regression on another function, where the safest way seems to be even more generated tests.

Comment (333fred) [reply src/Compilers/CSharp/Portable/Syntax/StackAllocArrayCreationExpressionSyntax.cs:12]: Most of these update functions are generated, and we only handwrite backcompat overloads. We apparently missed adding a test for this one. Please add a test.

**Review by 333fred — APPROVED** [review]
@dotnet/roslyn-compiler for a second review of this community pr.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by jcouv** [comment]
Merged/squashed. Thanks @AlFasGD for the fix

**Pull request merged**